### PR TITLE
feat: save custom metric percent format

### PR DESCRIPTION
**Pull request by almeidabbm**
Relates to: #8688 

### Description:

- Adds prefix, suffix, separator and currency columns to `'saved_queries_version_additional_metrics'` table.
- Saves and displays custom metric percent format.
- Improvements to typings.
- `convertAdditionalMetric` function returns new saved properties for custom metrics ( prefix, suffix and separator )
- renames types on formatting (for both Table Calculations and Custom metrics) so these are shared

Formatting settings that currently work:
- Round
- Compact
- Percent

Missing:
- Prefix
- Suffix
- Separator
- Currency

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/common/src/types/field.ts [hunk 1]
```diff
@@ -395,7 +395,10 @@ export const isMetric = (field: Field | undefined): field is Metric =>
 export const isNonAggregateMetric = (field: Field): boolean =>
     isMetric(field) && NonAggregateMetricTypes.includes(field.type);
 
-export interface Metric extends Field {
+export interface Metric
+    extends Field,
+        Pick<CustomFormat, 'prefix' | 'suffix' | 'separator'>,
+        Pick<AdditionalMetric, 'baseDimensionName'> {
```
Comment (joaoviana) [inline packages/common/src/types/field.ts:401]: we should remove this bit ^

File: packages/common/src/utils/formatting.ts [hunk 2]
```diff
@@ -270,7 +284,7 @@ export function formatFieldValue(
         case MetricType.COUNT:
         case MetricType.COUNT_DISTINCT:
         case MetricType.SUM:
-            return formatValue(value, { format, round, compact });
+            return formatValue(value, formatOpts);
```
Comment (joaoviana) [inline packages/common/src/utils/formatting.ts:287]: let's bring the previous change back: 

```
return formatValue(value, { format, round, compact });
```

and remove the `isAdditionalMetric` check 👍

**Review by joaoviana — CHANGES_REQUESTED** [review]
2 changes requested 👍

File: packages/common/src/types/field.ts (hunk 1 above)
Comment (almeidabbm) [reply packages/common/src/types/field.ts]: Done!

File: packages/common/src/utils/formatting.ts (hunk 2 above)
Comment (almeidabbm) [reply packages/common/src/utils/formatting.ts]: Done as well 👌

**Review by joaoviana — APPROVED** [review]
Looks great!

**Pull request merged**